src/main/java/org/terasology/module/health/events/BeforeRegenEvent.java

```diff
@@ -4,17 +4,31 @@
 package org.terasology.module.health.events;
 
 import org.terasology.engine.entitySystem.event.AbstractConsumableValueModifiableEvent;
+import org.terasology.naming.Name;
 
+/**
+ * A <i>collector event</i> to allow interested systems to contribute to a regeneration action.
+ *
+ * A separate collector event is sent for each registered regeneration id. The resulting amount per regeneration id is
+ * at minimum 0 (no negative effects due to regeneration). The final regeneration amount is computed from the sum of
+ * all amounts per regeneration id.
+ *
+ * To inflict damage caused by regeneration send a separate {@link DoDamageEvent} instead.
+ */
 public class BeforeRegenEvent extends AbstractConsumableValueModifiableEvent {
 
-    private final String id;
+    final Name id;
 
-    protected BeforeRegenEvent(String id, float baseValue) {
+    public BeforeRegenEvent(Name id, float baseValue) {
```

Comment:
why is this not `protected`?

Reply:
I don't think protected does make much sense here, as it would allow sub-classes to access it - whatever we'd need a subclass for this event for. 

The main problem here is that I sorted event into an `events` package, and systems into a `systems` package. For the systems to be able to create these events they need to be visible to them, and in this case this means the event constructor needs to be public. 

Maybe it makes sense to give another package structure a try :thinking: I'm moving everything around anyways, so let's see how this would turn out...

**tl;dr:** it needs to be public to be accessible to classes in the `systems` package.

Reply:
Huh, didn't think about it that way... That might indeed be a valid insight into how we should structure packages... Looking forward to see what you'll come up with 👍